### PR TITLE
Move TCL test unit/geo to Go case

### LINKED ISSUE #860: Move TCL test unit/geo to Go case

### PR DESCRIPTION
**Pull request by tanruixiang**
This close #860

### CODE REVIEW

**Comment by tisonkun** [comment]
```
--- FAIL: TestReplicationWithLimitSpeed (10.10s)
    client.go:72: 
        	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/replication/client.go:72
        	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/replication/replication_test.go:209
        	Error:      	Received unexpected error:
        	            	i/o timeout
        	Test:       	TestReplicationWithLimitSpeed
```

Unstable test. @git-hulk perhaps release the read timeout for the specific case.

File: tests/gocase/unit/geo/geo_test.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+	})
+
+	t.Run("GEOPOS simple", func(t *testing.T) {
+		require.NoError(t, rdb.Del(ctx, "points").Err())
+		require.NoError(t, rdb.GeoAdd(ctx, "points", &redis.GeoLocation{Name: "a", Longitude: 10, Latitude: 20}, &redis.GeoLocation{Name: "b", Longitude: 30, Latitude: 40}).Err())
+		cmd := rdb.GeoPos(ctx, "points", "a", "b")
+		require.True(t, math.Abs(cmd.Val()[0].Longitude-10) < 0.001)
+		require.True(t, math.Abs(cmd.Val()[0].Latitude-20) < 0.001)
+		require.True(t, math.Abs(cmd.Val()[1].Longitude-30) < 0.001)
+		require.True(t, math.Abs(cmd.Val()[1].Latitude-40) < 0.001)
```
Comment (tisonkun) [inline tests/gocase/unit/geo/geo_test.go]: Use `require.Less`.

File: tests/gocase/unit/geo/geo_test.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+	})
+
+	t.Run("GEOPOS missing element", func(t *testing.T) {
+		require.NoError(t, rdb.Del(ctx, "points").Err())
+		require.NoError(t, rdb.GeoAdd(ctx, "points", &redis.GeoLocation{Name: "a", Longitude: 10, Latitude: 20}, &redis.GeoLocation{Name: "b", Longitude: 30, Latitude: 40}).Err())
+		require.EqualValues(t, (*redis.GeoPos)(nil), rdb.GeoPos(ctx, "points", "a", "x", "b").Val()[1])
```
Comment (tisonkun) [inline tests/gocase/unit/geo/geo_test.go]:
```suggestion
		require.Nil(t, rdb.GeoPos(ctx, "points", "a", "x", "b").Val()[1])
```

File: tests/gocase/unit/geo/geo_test.go [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+	t.Run("GEORADIUS simple (sorted)", func(t *testing.T) {
+		require.EqualValues(t, []redis.GeoLocation([]redis.GeoLocation{{Name: "central park n/q/r", Longitude: 0, Latitude: 0, Dist: 0, GeoHash: 0}, {Name: "4545", Longitude: 0, Latitude: 0, Dist: 0, GeoHash: 0}, {Name: "union square", Longitude: 0, Latitude: 0, Dist: 0, GeoHash: 0}}), rdb.GeoRadius(ctx, "nyc", -73.9798091, 40.7598464, &redis.GeoRadiusQuery{Radius: 3, Unit: "km", Sort: "asc"}).Val())
+	})
+
+	t.Run("GEORADIUS with COUNT", func(t *testing.T) {
+		require.EqualValues(t, []redis.GeoLocation([]redis.GeoLocation{{Name: "central park n/q/r", Longitude: 0, Latitude: 0, Dist: 0, GeoHash: 0}, {Name: "4545", Longitude: 0, Latitude: 0, Dist: 0, GeoHash: 0}, {Name: "union square", Longitude: 0, Latitude: 0, Dist: 0, GeoHash: 0}}), rdb.GeoRadius(ctx, "nyc", -73.9798091, 40.7598464, &redis.GeoRadiusQuery{Radius: 10, Unit: "km", Sort: "asc", Count: 3}).Val())
+	})
+
+	t.Run("GEORADIUS HUGE, issue #2767", func(t *testing.T) {
```
Comment (tisonkun) [inline tests/gocase/unit/geo/geo_test.go]:
```suggestion
	t.Run("GEORADIUS HUGE, (redis issue #2767)", func(t *testing.T) {
```

**Review by tisonkun — COMMENTED** [review]
Thanks for your contribution! Comments inline.

**Comment by tanruixiang** [comment]
@tisonkun Thanks for your review, it has been revised.

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your contribution!

Merging...

**Pull request merged**